util/patricia_trie/src/triedb.rs

```diff
@@ -39,45 +42,51 @@ use bytes::Bytes;
 /// use hashdb::*;
 /// use memorydb::*;
 /// use ethereum_types::H256;
+/// use node_codec::RlpNodeCodec;
+///
+/// type RlpCodec = RlpNodeCodec<KeccakHasher>;
 ///
 /// fn main() {
-///   let mut memdb = MemoryDB::new();
+///   let mut memdb = MemoryDB::<KeccakHasher>::new();
 ///   let mut root = H256::new();
-///   TrieDBMut::new(&mut memdb, &mut root).insert(b"foo", b"bar").unwrap();
-///   let t = TrieDB::new(&memdb, &root).unwrap();
+///   TrieDBMut::<_, RlpCodec>::new(&mut memdb, &mut root).insert(b"foo", b"bar").unwrap();
+///   let t = TrieDB::<_, RlpCodec>::new(&memdb, &root).unwrap();
 ///   assert!(t.contains(b"foo").unwrap());
 ///   assert_eq!(t.get(b"foo").unwrap().unwrap(), DBValue::from_slice(b"bar"));
 /// }
 /// ```
-pub struct TrieDB<'db> {
-	db: &'db HashDB,
-	root: &'db H256,
+pub struct TrieDB<'db, H, C>
```

Comment:
have you tried changing the order of generics?

```
struct TrieDB<'db, C, H> { }
```

I believe it will allow you to init `TrieDB` without boilerplate `_`.

`TrieDB::<_, RlpCodec>new` -> `TrieDB::<RlpCodec>::new`

Reply:
Sadly it doesn't. It's kind of dumb actually: seems like there's enough info in `NodeCodec<H>` to infer that the `H` param must be the same type. :(